packages/common/src/api/tan-query/usePurchases.ts

```diff
@@ -0,0 +1,104 @@
+import { useMemo } from 'react'
+
+import { full } from '@audius/sdk'
+import { useInfiniteQuery } from '@tanstack/react-query'
+
+import { Id } from '~/api/utils'
+import { useAudiusQueryContext } from '~/audius-query'
+import { ID } from '~/models'
+import {
+  USDCContentPurchaseType,
+  USDCPurchaseDetails
+} from '~/models/USDCTransactions'
+import { Nullable } from '~/utils/typeUtils'
+
+import { QUERY_KEYS } from './queryKeys'
+import { useCollections } from './useCollections'
+import { useTracks } from './useTracks'
+import { useUsers } from './useUsers'
+import { parsePurchase } from './utils/parsePurchase'
+
+export type GetPurchaseListArgs = {
+  userId: Nullable<ID>
+  sortMethod?: full.GetPurchasesSortMethodEnum
+  sortDirection?: full.GetPurchasesSortDirectionEnum
+}
+
+export type PaginatedPurchaseListArgs = GetPurchaseListArgs & {
+  limit: number
+  offset: number
+}
+
+export const usePurchases = (
+  args: GetPurchaseListArgs,
+  options: { pageSize: number; enabled?: boolean }
+) => {
+  const { audiusSdk } = useAudiusQueryContext()
+  const { pageSize, enabled } = options
+  const queryResult = useInfiniteQuery({
+    queryKey: [QUERY_KEYS.purchases, args],
```

Comment:
just checking queryKey allows objects and it is stable across re-renders?

Reply:
oh good call, I would not pass objects for this, but break out the parts of the args that should affect whether this needs to re-render